core/templates/dev/head/services/contextual/UrlService.js

```diff
@@ -50,14 +50,14 @@ oppia.factory('UrlService', ['$window', function($window) {
     // story editor, since both have topic id in their url.
     getTopicIdFromUrl: function() {
       var pathname = this.getPathname();
-      if (pathname.match(/\/(story|topic)_editor\/\w{12}\b/g)) {
+      if (pathname.match(/\/(story|topic)_editor\/(\w|-){12}\b/g)) {
```

Comment:
I think underscores are allowed too, btw, here and below.

Reply:
The \w character checks for underscores also.